common/persistence/client/fault_injection_errorgenerator.go

```diff
@@ -138,13 +138,13 @@ func (p *DefaultErrorGenerator) Generate() error {
 	p.Lock()
 	defer p.Unlock()
 
-	if roll := p.r.Float64(); roll < p.rate.Load() {
+	if roll := p.r.Float64(); roll < math.Float64frombits(p.rate.Load()) {
 		var result error
 		for _, fm := range p.faultMetadata {
 			if roll < fm.threshold {
 				msg := fmt.Sprintf(
 					"FaultInjectionGenerator. rate %f, roll: %f, treshold: %f",
-					p.rate.Load(),
+					math.Float64frombits(p.rate.Load()),
```

Comment:
You should load the value once before using it. Otherwise, there's a risk of the rate changing and the message in L147 showing a value not corresponding to the comparison in L141. Issue not introduced in this PR though.

Reply:
Fixed issue